storage/liveness.proto

```diff
@@ -0,0 +1,34 @@
+// Copyright 2016 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+// Author: Spencer Kimball (spencer@cockroachlabs.com)
+
+syntax = "proto3";
+package cockroach.storage;
+option go_package = "storage";
+
+import "cockroach/util/hlc/timestamp.proto";
+import "gogoproto/gogo.proto";
+
+// Liveness holds information about a node's latest heartbeat and epoch.
+message Liveness {
+  int32 node_id = 1 [(gogoproto.customname) = "NodeID",
+      (gogoproto.casttype) = "github.com/cockroachdb/cockroach/roachpb.NodeID"];
+  // Epoch is a monotonically-increasing value for node liveness. It
+  // may be incremented every time a node's last heartbeat is older
+  // than a threshold liveness duration.
+  int64 epoch = 2;
+  // The timestamp at which this liveness record expires.
+  util.hlc.Timestamp expiration = 3 [(gogoproto.nullable) = false];
```

Comment:
Ah, actually I think we do need to send the `MaxOffset` because the case I just mentioned only holds if Node B's `MaxOffset` is greater than or equal to Node A's. What we need is Node A to stop serving reads at `Expiration` and Node B to consider a lease or liveness record expired only after `Expiration + Max(MaxOffset[Node B], MaxOffset[Node A])`.

But again, this just is begging the question of whether we ought to support changing max offset without requiring a cluster freeze. Is this kind of complexity healthy when you weigh costs and benefits?